File: threads/src/main/java/im/threads/internal/opengraph/OGData.java
package im.threads.internal.opengraph;

import android.text.TextUtils;

public final class OGData {
    private String title;
    private String description;
    private String imageUrl;
    private String url;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public boolean isEmpty() {
        return TextUtils.isEmpty(title)
                && TextUtils.isEmpty(description)
                && TextUtils.isEmpty(imageUrl)
                && TextUtils.isEmpty(url);
    }

    public boolean areTextsEmpty() {
        return TextUtils.isEmpty(title)
                && TextUtils.isEmpty(description)
                && TextUtils.isEmpty(url);
    }

    public String toString() {
        return "Open Graph Data:" +
                "[title:" + title +
                ", desc:" + description +
                ", imageUrl:" + imageUrl +
                ", url:" + url + "]";
    }

}

File: threads/src/main/java/im/threads/internal/domain/ogParser/OGData.kt
package im.threads.internal.domain.ogParser

/**
 * Содержит набор параметров, возвращаемых после парсинга meta data у html.
 */
data class OGData(
    var title: String = "",
    var description: String = "",
    var imageUrl: String = "",
    var url: String = "",
    var siteName: String = "",
    var type: String = ""
) {
    fun isEmpty() = title.isEmpty() && description.isEmpty() && url.isEmpty()

    override fun toString(): String {
        return "Open Graph Data:[title:$title, desc: $description, imageUrl: imageUrl, url:$url," +
            " siteName: $siteName, type: $type]"
    }
}
